src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+                frame = frame_count.GetFrame(0, 0)
+                data = np.reshape(np.fromiter(frame.GetData(), 'uint16'), [frame.Width, frame.Height], order='F')
+            else:
+                data = np.array([])
+                for i in range(0, frame_count.Frames):
+                    frame = frame_count.GetFrame(0, i)
+                    new_frame = np.fromiter(frame.GetData(), 'uint16')
+                    new_frame = np.reshape(np.fromiter(frame.GetData(), 'uint16'), [frame.Width, frame.Height],order='F')
+                    data = np.dstack((data, new_frame)) if data.size else new_frame
+            return data
+        else:
+            logger.warning('frame_count.Regions is not valid. Please retry.')
+            logger.info('Frame count: %s', frame_count.Frames)
+        return np.array([[]])  # Return an empty 2D numpy array by default
+
+    def close(self):
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self.automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+
+class Spectrometer():
+
+    def __init__(self, experiment_name=None):
+        self._experiment_name = experiment_name
+        self.light = LightfieldApp(True)
+
+    def finalize(self):
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self.light.close()
+
+    @property
+    def center_wavelength(self):
+        """
+        Returns the center wavelength in nanometers.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingCenterWavelength)
+
+    def get_wavelengths(self):
+        """
+        Returns the wavelength calibration for a single frame.
+        """
+        size = self.light.experiment.SystemColumnCalibration.get_Length()
+        result = np.empty(size)
+        for i in range(size):
+            result[i] = self.light.experiment.SystemColumnCalibration[i]
+        return result
+
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers):
+        """
+        Sets the spectrometer center wavelength to nanometers.
+        """
+        #NOTE: The line below addresses bug where if step and glue is selected, doesn't allow setting center wavelength
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+        return self.light.set(lf.AddIns.SpectrometerSettings.GratingCenterWavelength, nanometers)
+
+    @property
+    def experiment_name(self):
+        """
+        Returns the experiment name.
+        """
+        return self._experiment_name
+
+    @experiment_name.setter
+    def experiment_name(self, a_name):
+        """
+        User can set the experiment that they want to load.
+        """
+        if a_name != self._experiment_name:
+            self._experiment_name = a_name
+            self.light.load_experiment(self._experiment_name)
+
+    @property
+    def grating(self):
+        """
```

Comment:
the grating string is a mystery to me. It would be great if you could provide more detail or URL.

Reply:
we chatted about this in Slack. I think a pull-down is a great idea. But we just need to make sure we understand how that's going to affect the data that the spectrometer generates.